pkg/cmd/roachtest/tests/ptp.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package tests
+
+import (
+	"context"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/option"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/registry"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/install"
+	"github.com/stretchr/testify/require"
+	"strings"
+)
+
+func registerPTP(r registry.Registry) {
+
+	r.Add(registry.TestSpec{
+		Name:             "ptp",
+		Owner:            registry.OwnerServer, // TODO(tbg): KV?
+		Cluster:          r.MakeClusterSpec(1), // TODO(tbg): 3?
+		Leases:           registry.MetamorphicLeases,
+		Suites:           registry.Suites(registry.Weekly),
+		CompatibleClouds: registry.OnlyGCE, // TODO(tbg)
+		Run: func(ctx context.Context, t test.Test, c cluster.Cluster) {
+			require.False(t, c.IsLocal())
+			// TODO(tbg): do some verification in each step, i.e. at least a few SQL queries
+			// but probably a minute of running kv95 or something like that while using PTP.
+			// Or, emit proper benchmark artifact from the PTP run so we can see the overhead
+			// of using a ptp device vs vanilla.
+
+			firstOpt := install.DefaultRunOptions()
+			firstOpt.Nodes = c.Nodes(1).InstallNodes()
+			details, err := c.RunWithDetails(ctx, t.L(), firstOpt, "uname -r")
+			uname := strings.TrimSpace(details[0].Stdout)
+
+			_, err = c.RunWithDetails(ctx, t.L(), install.DefaultRunOptions(),
+				"sudo apt-get update -y && sudo apt-get install linux-modules-extra-"+uname+" && sudo modprobe ptp_kvm")
+			require.NoError(t, err)
```

Comment:
I've never quite understood the distinction, but we also have these "scripts" that we can "install".

E.g. [Calling `c.Install(...)`](https://github.com/cockroachdb/cockroach/blob/f80bc8a04b3f8c567af6bfdb719c1efc0fbe32f8/pkg/cmd/roachtest/cluster.go#L2598-L2609), one can grab things from [this map](https://github.com/cockroachdb/cockroach/blob/f80bc8a04b3f8c567af6bfdb719c1efc0fbe32f8/pkg/roachprod/install/install.go#L28).